### PR TITLE
Adding logging to lambda and making sure the permissions are OK.

### PR DESCRIPTION
**Pull request by adhorn**
*Issue #11   :*

*Description of changes:*
- Review and streamline the permissions
- adding logs to all lambda functions
- renaming StepFunction role
- Tested with full deploy

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 PublicWorkshops.xpr
 docs/Template/
 docs/
+.DS_Store
```
Comment (setheliot) [inline .gitignore]: was removing the LF here intentional?

File: Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/CloudFormation/lambda_functions_for_deploy.json [hunk 2]
```diff
@@ -180,26 +193,26 @@
...
       }
     },
 
     "ELBServiceRole" : {
-      "Type" : "AWS::IAM::ServiceLinkedRole",
-      "Condition" : "CreateELBServiceRole",
-      "Properties" : {
-        "AWSServiceName" : "elasticloadbalancing.amazonaws.com",
-        "Description"    : "This should allow an ELB to be created by this account"
+      "Type": "AWS::IAM::ServiceLinkedRole",
```
Comment (setheliot) [inline Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/CloudFormation/lambda_functions_for_deploy.json]: Is the only change here to remove the spaces before the colon?  I see that done on other changes too, but there are still plenty of places that were not updated (and still include the spaces).  Why fix here (if no other changes) but not elsewhere?  Or we can make the whole thing moot with https://github.com/setheliot/aws-well-architected-labs/issues/16 :-)

File: Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/CloudFormation/lambda_functions_for_deploy.json [hunk 3]
```diff
@@ -357,11 +376,16 @@
           } ]
         },
         "Policies" : [ {
-          "PolicyName" : "CreateVPC",
+          "PolicyName" : "StateMachineRole",
           "PolicyDocument" : {
             "Statement" : [ {
               "Effect" : "Allow",
-              "Action" :  "*",
+              "Action" :  [
```
Comment (setheliot) [inline Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/CloudFormation/lambda_functions_for_deploy.json]: nice!

File: .gitignore (hunk 1 above)
Comment (adhorn) [reply .gitignore]: not on purpose, but not a breaker .. or is it?

File: Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/CloudFormation/lambda_functions_for_deploy.json (hunk 2 above)
Comment (adhorn) [reply Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/CloudFormation/lambda_functions_for_deploy.json]: Yea - part of working on code is cleaning it. Updating parts that I touch only.

File: .gitignore (hunk 1 above)
Comment (adhorn) [reply .gitignore]: fix now in https://github.com/setheliot/aws-well-architected-labs/pull/12/commits/fe6255e90057d9f153ff6a7eb2e57f074c613698

Comment (setheliot) [reply .gitignore]: I did not know if it was a breaker... just noticed it.   
Dumb Git question: I see you added that commit to https://github.com/setheliot/aws-well-architected-labs/pull/12, but do you also need to add the other PRs?  If you do not won't the other PRs "undo"  https://github.com/setheliot/aws-well-architected-labs/pull/1?

**Comment by setheliot** [comment]
The good news is we know the Lambda logging to CW works!  So I will pull https://github.com/setheliot/aws-well-architected-labs/pull/12
image

The bad news is that I can see in those CW logs a bug in https://github.com/setheliot/aws-well-architected-labs/pull/15 ;-)

**Pull request merged**